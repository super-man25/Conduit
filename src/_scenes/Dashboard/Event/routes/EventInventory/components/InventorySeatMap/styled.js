import React from 'react';
import { Box, CenteredLoader } from '_components';
import styled from 'styled-components';

export const EventInventoryVerticalRule = styled(Box)`
  background-color: #979797;
  cursor: pointer;
`;

export const SeatIcon = styled.img`
  cursor: pointer;
`;

export const SeatMapLoader = () => (
  <div style={{ position: 'relative', height: '100%', minHeight: 400 }}>
    <CenteredLoader />
  </div>
);
