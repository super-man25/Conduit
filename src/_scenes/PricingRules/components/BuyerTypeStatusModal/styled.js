import styled from 'styled-components';
import { cssConstants } from '_constants';
import { H4, Flex, Input, Text } from '_components';
import { darken } from 'polished';

export const ModalOverlay = styled.div`
  position: fixed;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  opacity: 0.4;
  background-color: ${cssConstants.PRIMARY_LIGHT_BLACK};
`;

export const ModalContent = styled.div`
  position: fixed;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
  width: 500px;
  z-index: 9;
  box-shadow: 0 1px 10px rgba(0, 0, 0, 0.27);
`;

export const ModalHeader = styled.div`
  padding: 1.5rem;
  background-color: ${cssConstants.PRIMARY_LIGHTEST_GRAY};
  border-bottom: 1px solid ${cssConstants.PRIMARY_DARK_BLUE};
  border-top-left-radius: 3px;
  border-top-right-radius: 3px;
`;

export const ModalBody = styled.div`
  padding: 1rem 2rem;
  height: 400px;
  overflow-x: hidden;
  overflow-y: scroll;
  background-color: ${cssConstants.PRIMARY_LIGHTEST_GRAY};
  border-bottom: 1px solid ${cssConstants.PRIMARY_DARK_BLUE};
`;

export const ModalFooter = styled.div`
  padding: 1.5rem;
  background-color: ${cssConstants.PRIMARY_LIGHTEST_GRAY};
  border-bottom: ${cssConstants.PRIMARY_DARK_BLUE};
  border-bottom-left-radius: 3px;
  border-bottom-right-radius: 3px;
`;

export const BuyerTypeOption = Flex.extend`
  padding: 0 2rem;
  min-height: 45px;
  width: 100%;
  background-color: ${cssConstants.PRIMARY_LIGHTEST_GRAY};

  border-bottom: 1px solid ${cssConstants.PRIMARY_LIGHTER_GRAY};

  :hover {
    background-color: ${darken(0.05, cssConstants.PRIMARY_WHITE)};
  }
`;

export const FlexText = Text.extend`
  flex-basis: ${(props) => props.basis || '10%'};
`;

export const ErrorText = Text.extend`
  text-align: right;
`;

export const Title = H4.extend`
  margin: 0;
  margin-bottom: 2rem;
  color: ${cssConstants.PRIMARY_LIGHT_BLACK};
`;

export const FilterBuyerTypesInput = Input.extend`
  border-radius: 20px;
  height: 35px;
  width: 80px;
`;
