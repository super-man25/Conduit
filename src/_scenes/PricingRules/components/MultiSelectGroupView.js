// @flow

import styled from 'styled-components';
import * as React from 'react';
import { Text, Flex } from '_components';
import { Checkbox, MultiSelectOption } from './styled';
import { colors } from '_constants';

const MultiSelectOptionHeader = styled(MultiSelectOption)`
  background-color: ${colors.blue};
  color: #fff;

  :hover {
    background-color: ${colors.blue};
  }
`;

type Props = {
  categories: any[],
  grouped: any,
  selected: any[],
  labelFn: (option: any) => string,
  onItemClicked: (option: any) => void,
  updatePriceRuleProperty: (options: any[]) => void,
};

type State = {
  selectAllNext: any,
};

export class MultiSelectGroupView extends React.Component<Props, State> {
  state = { selectAllNext: {} };

  componentDidMount() {
    const selectAllNext = this.props.categories.reduce(
      (acc, category) => ({
        ...acc,
        [category.id]: true,
      }),
      {}
    );

    this.setState({ selectAllNext });
  }

  toggleSelectAllCategory(category: any) {
    const { updatePriceRuleProperty, selected, grouped } = this.props;
    const { selectAllNext } = this.state;

    let selectedIds;
    if (selectAllNext[category.id]) {
      selectedIds = grouped[category.id]
        .map((e) => e.id)
        .concat(selected)
        .filter((event, index, events) => events.indexOf(event) === index);
    } else {
      const idsInGroup = grouped[category.id].map((o) => o.id);
      selectedIds = selected.filter((s) => !idsInGroup.includes(s));
    }

    updatePriceRuleProperty(selectedIds);
    this.setState({
      selectAllNext: {
        ...selectAllNext,
        [category.id]: !selectAllNext[category.id],
      },
    });
  }

  render() {
    const {
      categories,
      grouped,
      selected,
      onItemClicked,
      labelFn,
    } = this.props;

    return (
      <React.Fragment>
        {categories.map((category) => {
          return (
            <React.Fragment key={`category${category.id}`}>
              <MultiSelectOptionHeader>
                <Flex align="center" justify="space-between">
                  {category.name}
                  <Text
                    color="#fff"
                    style={{ cursor: 'pointer' }}
                    onClick={() => this.toggleSelectAllCategory(category)}
                  >
                    {this.state.selectAllNext[category.id]
                      ? 'Select All'
                      : 'Select None'}
                  </Text>
                </Flex>
              </MultiSelectOptionHeader>
              {grouped[category.id].map((option) => {
                return (
                  <MultiSelectOption
                    key={option.id}
                    isActive={selected.includes(option.id)}
                    onClick={() => onItemClicked(option)}
                  >
                    <Flex align="center">
                      <Checkbox
                        checked={selected.includes(option.id)}
                        onChange={() => onItemClicked(option)}
                      />
                      {labelFn(option)}
                    </Flex>
                  </MultiSelectOption>
                );
              })}
            </React.Fragment>
          );
        })}
      </React.Fragment>
    );
  }
}
