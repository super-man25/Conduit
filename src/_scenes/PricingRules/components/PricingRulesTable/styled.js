import styled from 'styled-components';
import { cssConstants } from '_constants';
import { PrimaryButton, Input, Box, SecondaryButton } from '_components';

export const EditPricingRuleInput = Input.extend`
  box-sizing: border-box;
  margin: 0 0.5rem 0 0;
  padding: 10px 8px;
  max-height: 35px;
  max-width: 80px;
`;

export const SavePricingRuleButton = PrimaryButton.extend`
  box-sizing: border-box;
  margin: 0;
  min-width: 50px;
  max-width: 75px;
  max-height: 35px;
`;

export const CancelEditingButton = SecondaryButton.extend`
  box-sizing: border-box;
  margin: 0;
  min-width: 50px;
  max-width: 75px;
  max-height: 35px;
  padding: 0 0.75rem;
`;

export const TableHeaderCell = styled.div`
  text-align: ${(props) => props.align || 'left'};
  color: ${cssConstants.PRIMARY_DARKEST_GRAY};
  font-weight: 500;
`;

export const PositionedBox = Box.extend`
  position: ${(props) => props.position};
  top: ${(props) => props.top};
  bottom: ${(props) => props.bottom};
  left: ${(props) => props.left};
  right: ${(props) => props.right};
  z-index: ${(props) => props.zIndex};
`;
export const Checkbox = styled.input.attrs({ type: 'checkbox' })`
  margin-right: 0.5rem;
`;
