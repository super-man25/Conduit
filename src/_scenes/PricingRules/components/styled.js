import styled, { css } from 'styled-components';

import { colors } from '_constants';
import { withClickAway } from '_hoc';
import { Input, Box } from '_components';

export const EditPricingRuleInput = styled(Input)`
  margin: 0 0.5rem 0 0;
  padding: 10px 8px;
  max-height: 35px;
  max-width: 80px;
`;

export const TableHeaderCell = styled.div`
  font-weight: bold;
`;

export const PositionedBox = styled(Box)`
  position: ${(props) => props.position};
  top: ${(props) => props.top};
  bottom: ${(props) => props.bottom};
  left: ${(props) => props.left};
  right: ${(props) => props.right};
  z-index: ${(props) => props.zIndex};
`;
export const Checkbox = styled.input.attrs(() => ({ type: 'checkbox' }))`
  margin-right: 0.5rem;
`;

export const MultiSelectContainer = withClickAway(styled.div`
  position: relative;
  cursor: pointer;
  display: flex;
  justify-content: space-between;
  align-items: center;
`);

export const MultiSelectMenu = styled.div`
  position: absolute;
  width: 250px;
  top: calc(100% + 8px);
  max-height: 350px;
  overflow-y: scroll;
  overflow-x: hidden;
  z-index: 10;
  border: 1px solid ${colors.lightGray};
  box-shadow: 0 20px 20px rgba(0, 0, 0, 0.06);

  transition: 0.1s ease-in-out all;
  opacity: 0;
  transform: translateY(20px);
  visibility: hidden;

  ${(props) =>
    props.show &&
    css`
      opacity: 1;
      transform: translateY(0);
      visibility: visible;
    `}
`;

export const MultiSelectOption = styled.div`
  padding: 12px 16px;
  background-color: ${(props) =>
    props.isActive ? colors.lightGray : colors.white};
  transition: 0.1s ease-in-out all;
  color: ${colors.black};
  white-space: normal;

  :not(:last-child) {
    border-bottom: 1px solid ${colors.lightGray};
  }

  :hover {
    cursor: ${(props) => (props.isActive ? 'default' : 'pointer')};
    background-color: ${colors.lightGray};
  }
`;

export const SplitButtonContainer = styled.div`
  flex-basis: 100%;
  display: flex;
  justify-content: center;
  margin-top: 15px;
`;

export const SplitButtonHalf = styled.span`
  padding: 8px;
  border: 1px solid ${colors.blue};
  display: inline-block;
  min-width: 35%;
  text-align: center;
  cursor: pointer;
  color: ${(props) => (props.isActive ? colors.white : colors.blue)};
  background-color: ${(props) => (props.isActive ? colors.blue : colors.white)};

  :not(:last-child) {
    border-bottom-left-radius: 2px;
    border-top-left-radius: 2px;
    border-right: none;
  }

  :last-child {
    border-bottom-right-radius: 2px;
    border-top-right-radius: 2px;
  }
`;

export const DropdownFilterInputContainer = styled.div`
  background-color: white;
  padding: 10px;
  border-bottom: 1px solid ${colors.lightGray};
`;

export const DropdownFilterInput = styled.input`
  width: 100%;
  padding: 10px 15px;
  border: 1px solid ${colors.lightGray};
  border-radius: 3px;
`;
