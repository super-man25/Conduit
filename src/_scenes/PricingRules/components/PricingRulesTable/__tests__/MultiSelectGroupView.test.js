import React from 'react';
import { shallow, mount } from 'enzyme';
import { MultiSelectGroupView } from '../MultiSelectGroupView';
import { Checkbox } from '../styled';

describe('<MultiSelectGroupView />', () => {
  const props = {
    categories: [{ id: 2, name: 'Value' }, { id: 3, name: 'Classic' }],
    grouped: {
      2: [
        { id: 6, name: 'Braves at Zephyrs' },
        { id: 18, name: 'Dogs FIGHT Cats' }
      ],
      3: [{ id: 23, name: 'Cardinals at Zephyrs' }]
    },
    selected: [18],
    labelFn: (o) => o.name,
    onItemClicked: jest.fn(),
    updatePriceRuleProperty: jest.fn()
  };

  it('should render correctly', () => {
    const wrapper = shallow(<MultiSelectGroupView {...props} />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.state()).toEqual({
      selectAllNext: {
        2: true,
        3: true
      }
    });
  });

  it('should call update price rule with all events from category', () => {
    const fn = jest.fn();
    const wrapper = mount(
      <MultiSelectGroupView {...props} updatePriceRuleProperty={fn} />
    );

    wrapper
      .find('p')
      .at(0)
      .simulate('click');

    expect(fn).toBeCalled();
    expect(fn.mock.calls[0][0]).toEqual([6, 18]);
    expect(wrapper.state().selectAllNext).toEqual({ 2: false, 3: true });

    wrapper.unmount();
  });

  it('should call onItemClicked when an item is clicked', () => {
    const fn = jest.fn();
    const wrapper = shallow(
      <MultiSelectGroupView {...props} onItemClicked={fn} />
    );

    wrapper
      .find(Checkbox)
      .at(0)
      .simulate('change');

    expect(fn).toBeCalled();
  });
});
