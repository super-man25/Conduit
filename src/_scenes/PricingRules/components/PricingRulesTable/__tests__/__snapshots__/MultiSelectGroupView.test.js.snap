// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MultiSelectGroupView /> should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MultiSelectGroupView
    categories={
        Array [
            Object {
              "id": 2,
              "name": "Value",
            },
            Object {
              "id": 3,
              "name": "Classic",
            },
          ]
    }
    grouped={
        Object {
            "2": Array [
              Object {
                "id": 6,
                "name": "Braves at Zephyrs",
              },
              Object {
                "id": 18,
                "name": "Dogs FIGHT Cats",
              },
            ],
            "3": Array [
              Object {
                "id": 23,
                "name": "Cardinals at Zephyrs",
              },
            ],
          }
    }
    labelFn={[Function]}
    onItemClicked={[Function]}
    selected={
        Array [
            18,
          ]
    }
    updatePriceRuleProperty={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <Unknown>
          <styled.div>
                    <Flex
                              align="center"
                              justify="space-between"
                    >
                              Value
                              <styled.p
                                        color="#fff"
                                        onClick={[Function]}
                                        style={
                                                  Object {
                                                            "cursor": "pointer",
                                                          }
                                        }
                              >
                                        Select All
                              </styled.p>
                    </Flex>
          </styled.div>
          <styled.div
                    isActive={false}
                    onClick={[Function]}
          >
                    <Flex
                              align="center"
                    >
                              <styled.input
                                        checked={false}
                                        onChange={[Function]}
                              />
                              Braves at Zephyrs
                    </Flex>
          </styled.div>
          <styled.div
                    isActive={true}
                    onClick={[Function]}
          >
                    <Flex
                              align="center"
                    >
                              <styled.input
                                        checked={true}
                                        onChange={[Function]}
                              />
                              Dogs FIGHT Cats
                    </Flex>
          </styled.div>
</Unknown>,
        <Unknown>
          <styled.div>
                    <Flex
                              align="center"
                              justify="space-between"
                    >
                              Classic
                              <styled.p
                                        color="#fff"
                                        onClick={[Function]}
                                        style={
                                                  Object {
                                                            "cursor": "pointer",
                                                          }
                                        }
                              >
                                        Select All
                              </styled.p>
                    </Flex>
          </styled.div>
          <styled.div
                    isActive={false}
                    onClick={[Function]}
          >
                    <Flex
                              align="center"
                    >
                              <styled.input
                                        checked={false}
                                        onChange={[Function]}
                              />
                              Cardinals at Zephyrs
                    </Flex>
          </styled.div>
</Unknown>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "category2",
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <styled.div>
              <Flex
                            align="center"
                            justify="space-between"
              >
                            Value
                            <styled.p
                                          color="#fff"
                                          onClick={[Function]}
                                          style={
                                                        Object {
                                                                      "cursor": "pointer",
                                                                    }
                                          }
                            >
                                          Select All
                            </styled.p>
              </Flex>
</styled.div>,
            Array [
              <styled.div
                isActive={false}
                onClick={[Function]}
>
                <Flex
                                align="center"
                >
                                <styled.input
                                                checked={false}
                                                onChange={[Function]}
                                />
                                Braves at Zephyrs
                </Flex>
</styled.div>,
              <styled.div
                isActive={true}
                onClick={[Function]}
>
                <Flex
                                align="center"
                >
                                <styled.input
                                                checked={true}
                                                onChange={[Function]}
                                />
                                Dogs FIGHT Cats
                </Flex>
</styled.div>,
            ],
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <Flex
                align="center"
                justify="space-between"
>
                Value
                <styled.p
                                color="#fff"
                                onClick={[Function]}
                                style={
                                                Object {
                                                                "cursor": "pointer",
                                                              }
                                }
                >
                                Select All
                </styled.p>
</Flex>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "align": "center",
                "children": Array [
                  "Value",
                  <styled.p
                    color="#fff"
                    onClick={[Function]}
                    style={
                                        Object {
                                                            "cursor": "pointer",
                                                          }
                    }
>
                    Select All
</styled.p>,
                ],
                "justify": "space-between",
              },
              "ref": null,
              "rendered": Array [
                "Value",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Select All",
                    "color": "#fff",
                    "onClick": [Function],
                    "style": Object {
                      "cursor": "pointer",
                    },
                  },
                  "ref": null,
                  "rendered": "Select All",
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "6",
            "nodeType": "class",
            "props": Object {
              "children": <Flex
                align="center"
>
                <styled.input
                                checked={false}
                                onChange={[Function]}
                />
                Braves at Zephyrs
</Flex>,
              "isActive": false,
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "align": "center",
                "children": Array [
                  <styled.input
                    checked={false}
                    onChange={[Function]}
/>,
                  "Braves at Zephyrs",
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "checked": false,
                    "onChange": [Function],
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "Braves at Zephyrs",
              ],
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "18",
            "nodeType": "class",
            "props": Object {
              "children": <Flex
                align="center"
>
                <styled.input
                                checked={true}
                                onChange={[Function]}
                />
                Dogs FIGHT Cats
</Flex>,
              "isActive": true,
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "align": "center",
                "children": Array [
                  <styled.input
                    checked={true}
                    onChange={[Function]}
/>,
                  "Dogs FIGHT Cats",
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "checked": true,
                    "onChange": [Function],
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "Dogs FIGHT Cats",
              ],
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": Symbol(react.fragment),
      },
      Object {
        "instance": null,
        "key": "category3",
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <styled.div>
              <Flex
                            align="center"
                            justify="space-between"
              >
                            Classic
                            <styled.p
                                          color="#fff"
                                          onClick={[Function]}
                                          style={
                                                        Object {
                                                                      "cursor": "pointer",
                                                                    }
                                          }
                            >
                                          Select All
                            </styled.p>
              </Flex>
</styled.div>,
            Array [
              <styled.div
                isActive={false}
                onClick={[Function]}
>
                <Flex
                                align="center"
                >
                                <styled.input
                                                checked={false}
                                                onChange={[Function]}
                                />
                                Cardinals at Zephyrs
                </Flex>
</styled.div>,
            ],
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <Flex
                align="center"
                justify="space-between"
>
                Classic
                <styled.p
                                color="#fff"
                                onClick={[Function]}
                                style={
                                                Object {
                                                                "cursor": "pointer",
                                                              }
                                }
                >
                                Select All
                </styled.p>
</Flex>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "align": "center",
                "children": Array [
                  "Classic",
                  <styled.p
                    color="#fff"
                    onClick={[Function]}
                    style={
                                        Object {
                                                            "cursor": "pointer",
                                                          }
                    }
>
                    Select All
</styled.p>,
                ],
                "justify": "space-between",
              },
              "ref": null,
              "rendered": Array [
                "Classic",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Select All",
                    "color": "#fff",
                    "onClick": [Function],
                    "style": Object {
                      "cursor": "pointer",
                    },
                  },
                  "ref": null,
                  "rendered": "Select All",
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "23",
            "nodeType": "class",
            "props": Object {
              "children": <Flex
                align="center"
>
                <styled.input
                                checked={false}
                                onChange={[Function]}
                />
                Cardinals at Zephyrs
</Flex>,
              "isActive": false,
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "align": "center",
                "children": Array [
                  <styled.input
                    checked={false}
                    onChange={[Function]}
/>,
                  "Cardinals at Zephyrs",
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "checked": false,
                    "onChange": [Function],
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "Cardinals at Zephyrs",
              ],
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": Symbol(react.fragment),
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <Unknown>
            <styled.div>
                        <Flex
                                    align="center"
                                    justify="space-between"
                        >
                                    Value
                                    <styled.p
                                                color="#fff"
                                                onClick={[Function]}
                                                style={
                                                            Object {
                                                                        "cursor": "pointer",
                                                                      }
                                                }
                                    >
                                                Select All
                                    </styled.p>
                        </Flex>
            </styled.div>
            <styled.div
                        isActive={false}
                        onClick={[Function]}
            >
                        <Flex
                                    align="center"
                        >
                                    <styled.input
                                                checked={false}
                                                onChange={[Function]}
                                    />
                                    Braves at Zephyrs
                        </Flex>
            </styled.div>
            <styled.div
                        isActive={true}
                        onClick={[Function]}
            >
                        <Flex
                                    align="center"
                        >
                                    <styled.input
                                                checked={true}
                                                onChange={[Function]}
                                    />
                                    Dogs FIGHT Cats
                        </Flex>
            </styled.div>
</Unknown>,
          <Unknown>
            <styled.div>
                        <Flex
                                    align="center"
                                    justify="space-between"
                        >
                                    Classic
                                    <styled.p
                                                color="#fff"
                                                onClick={[Function]}
                                                style={
                                                            Object {
                                                                        "cursor": "pointer",
                                                                      }
                                                }
                                    >
                                                Select All
                                    </styled.p>
                        </Flex>
            </styled.div>
            <styled.div
                        isActive={false}
                        onClick={[Function]}
            >
                        <Flex
                                    align="center"
                        >
                                    <styled.input
                                                checked={false}
                                                onChange={[Function]}
                                    />
                                    Cardinals at Zephyrs
                        </Flex>
            </styled.div>
</Unknown>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "category2",
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <styled.div>
                <Flex
                                align="center"
                                justify="space-between"
                >
                                Value
                                <styled.p
                                                color="#fff"
                                                onClick={[Function]}
                                                style={
                                                                Object {
                                                                                "cursor": "pointer",
                                                                              }
                                                }
                                >
                                                Select All
                                </styled.p>
                </Flex>
</styled.div>,
              Array [
                <styled.div
                  isActive={false}
                  onClick={[Function]}
>
                  <Flex
                                    align="center"
                  >
                                    <styled.input
                                                      checked={false}
                                                      onChange={[Function]}
                                    />
                                    Braves at Zephyrs
                  </Flex>
</styled.div>,
                <styled.div
                  isActive={true}
                  onClick={[Function]}
>
                  <Flex
                                    align="center"
                  >
                                    <styled.input
                                                      checked={true}
                                                      onChange={[Function]}
                                    />
                                    Dogs FIGHT Cats
                  </Flex>
</styled.div>,
              ],
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <Flex
                  align="center"
                  justify="space-between"
>
                  Value
                  <styled.p
                                    color="#fff"
                                    onClick={[Function]}
                                    style={
                                                      Object {
                                                                        "cursor": "pointer",
                                                                      }
                                    }
                  >
                                    Select All
                  </styled.p>
</Flex>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "align": "center",
                  "children": Array [
                    "Value",
                    <styled.p
                      color="#fff"
                      onClick={[Function]}
                      style={
                                            Object {
                                                                  "cursor": "pointer",
                                                                }
                      }
>
                      Select All
</styled.p>,
                  ],
                  "justify": "space-between",
                },
                "ref": null,
                "rendered": Array [
                  "Value",
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": "Select All",
                      "color": "#fff",
                      "onClick": [Function],
                      "style": Object {
                        "cursor": "pointer",
                      },
                    },
                    "ref": null,
                    "rendered": "Select All",
                    "type": [Function],
                  },
                ],
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "6",
              "nodeType": "class",
              "props": Object {
                "children": <Flex
                  align="center"
>
                  <styled.input
                                    checked={false}
                                    onChange={[Function]}
                  />
                  Braves at Zephyrs
</Flex>,
                "isActive": false,
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "align": "center",
                  "children": Array [
                    <styled.input
                      checked={false}
                      onChange={[Function]}
/>,
                    "Braves at Zephyrs",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "checked": false,
                      "onChange": [Function],
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  "Braves at Zephyrs",
                ],
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "18",
              "nodeType": "class",
              "props": Object {
                "children": <Flex
                  align="center"
>
                  <styled.input
                                    checked={true}
                                    onChange={[Function]}
                  />
                  Dogs FIGHT Cats
</Flex>,
                "isActive": true,
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "align": "center",
                  "children": Array [
                    <styled.input
                      checked={true}
                      onChange={[Function]}
/>,
                    "Dogs FIGHT Cats",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "checked": true,
                      "onChange": [Function],
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  "Dogs FIGHT Cats",
                ],
                "type": [Function],
              },
              "type": [Function],
            },
          ],
          "type": Symbol(react.fragment),
        },
        Object {
          "instance": null,
          "key": "category3",
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <styled.div>
                <Flex
                                align="center"
                                justify="space-between"
                >
                                Classic
                                <styled.p
                                                color="#fff"
                                                onClick={[Function]}
                                                style={
                                                                Object {
                                                                                "cursor": "pointer",
                                                                              }
                                                }
                                >
                                                Select All
                                </styled.p>
                </Flex>
</styled.div>,
              Array [
                <styled.div
                  isActive={false}
                  onClick={[Function]}
>
                  <Flex
                                    align="center"
                  >
                                    <styled.input
                                                      checked={false}
                                                      onChange={[Function]}
                                    />
                                    Cardinals at Zephyrs
                  </Flex>
</styled.div>,
              ],
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <Flex
                  align="center"
                  justify="space-between"
>
                  Classic
                  <styled.p
                                    color="#fff"
                                    onClick={[Function]}
                                    style={
                                                      Object {
                                                                        "cursor": "pointer",
                                                                      }
                                    }
                  >
                                    Select All
                  </styled.p>
</Flex>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "align": "center",
                  "children": Array [
                    "Classic",
                    <styled.p
                      color="#fff"
                      onClick={[Function]}
                      style={
                                            Object {
                                                                  "cursor": "pointer",
                                                                }
                      }
>
                      Select All
</styled.p>,
                  ],
                  "justify": "space-between",
                },
                "ref": null,
                "rendered": Array [
                  "Classic",
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": "Select All",
                      "color": "#fff",
                      "onClick": [Function],
                      "style": Object {
                        "cursor": "pointer",
                      },
                    },
                    "ref": null,
                    "rendered": "Select All",
                    "type": [Function],
                  },
                ],
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "23",
              "nodeType": "class",
              "props": Object {
                "children": <Flex
                  align="center"
>
                  <styled.input
                                    checked={false}
                                    onChange={[Function]}
                  />
                  Cardinals at Zephyrs
</Flex>,
                "isActive": false,
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "align": "center",
                  "children": Array [
                    <styled.input
                      checked={false}
                      onChange={[Function]}
/>,
                    "Cardinals at Zephyrs",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "checked": false,
                      "onChange": [Function],
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  "Cardinals at Zephyrs",
                ],
                "type": [Function],
              },
              "type": [Function],
            },
          ],
          "type": Symbol(react.fragment),
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
